### PR TITLE
Megacheck fixes

### PR DESCRIPTION
**Pull request by klauern**
Using [`megacheck`](https://github.com/dominikh/go-tools/tree/master/cmd/megacheck) - Run [staticcheck](https://github.com/dominikh/go-tools/tree/master/cmd/staticcheck), [gosimple](https://github.com/dominikh/go-tools/tree/master/cmd/gosimple) and [unused](https://github.com/dominikh/go-tools/tree/master/cmd/unused), sharing work.

These address some of the findings in `megacheck`.

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -134,16 +134,6 @@ func (r *Requester) SetClient(client *http.Client) *Requester {
 	return r
 }
 
-func (r *Requester) parseQueryString(queries map[string]string) string {
```
Comment (supereagle) [inline request.go]: Why this unused function is not found by `deadcode` tool in PR #95?

Comment (klauern) [inline request.go]: In looking at the [`README`](https://github.com/remyoudompheng/go-misc/tree/master/deadcode), it says the following:

# Limitations
* Self-referential unused code is not currently reported
* A single package can be tested at a time
* Unused methods are not reported

Comment (klauern) [inline request.go]: Here's another issue that talks about the same kind of thing:  alecthomas/gometalinter#98

**Comment by bndr** [comment]
LGTM

**Comment by supereagle** [comment]
Conflict with #96.  @klauern  Please rebase, thanks.

**Comment by klauern** [comment]
Done

**Comment by bndr** [comment]
Thanks for the PR!

**Pull request merged**